### PR TITLE
fix: ignore trailing slash in static payloads manifest (#8794)

### PR DESCRIPTION
**Pull request by danielroe**
## Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)

## Checklist:
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/generator/src/generator.js [hunk 1]
```diff
@@ -349,7 +350,7 @@ export default class Generator {
         }
         // Add route to manifest (only if no error and redirect)
         if (this.manifest && (!res.error && !res.redirected)) {
-          this.manifest.routes.push(route)
+          this.manifest.routes.push(withoutTrailingSlash(route))
```
Comment (pi0) [inline packages/generator/src/generator.js:353]: I think we are missing `|| '/'` fallback here. It should be finally fixed back in ufo :D

File: packages/vue-app/template/App.js [hunk 2]
```diff
@@ -328,14 +324,14 @@ export default {
       <% if (nuxtOptions.generate.manifest) { %>
       const manifest = await this.fetchStaticManifest()
       const path = this.getRoutePath(route)
-      if (!manifest.routes.includes(decodeURI(path))) {
+      if (!manifest.routes.includes(decode(path))) {
```
Comment (pi0) [inline packages/vue-app/template/App.js:327]: What if we normalize route once in line before? (decode and strip trailing-slash). This way we ensure consistency and little bit perf for not decoding multiple times...

Comment (danielroe) [reply packages/vue-app/template/App.js:327]: In fact it looks like there was a bug in payload fetching with routerBase that this fixes :see_no_evil:

File: packages/generator/src/generator.js (hunk 1 above)
Comment (pi0) [reply packages/generator/src/generator.js]: Added in 0.6.3

**Review by pi0 — APPROVED** [review]

**Pull request merged**